### PR TITLE
[move-mutator] if/else and delete operators 

### PR DESCRIPTION
**Pull request by asmie**
### Description

Two additional operators has been added:
- if/else replacement operator;
- statement delete operator.

`if/else` operator replaces expressions under the if statements into literals (true/false) and mutates the expression to the negated one ( (a) is mutated to !(a)). 
For example, the `if (cond) { ... } else { ... }` expression can be replaced with the `if (false) { ... } else { ... }` expression.

Delete operator replaces some statements (currently `move_to`) with empty statement.

### Test Plan
New Move assets has been added.

### CODE REVIEW

File: third_party/move/tools/move-mutator/src/operators/mod.rs [hunk 1]
```diff
@@ -1,4 +1,32 @@
...
+pub(crate) const MOVE_TRUE: &str = "true";
+pub(crate) const MOVE_FALSE: &str = "false";
+pub(crate) const MOVE_ZERO: &str = "0";
+pub(crate) const MOVE_ZERO_U8: &str = "0u8";
+pub(crate) const MOVE_ZERO_U16: &str = "0u16";
+pub(crate) const MOVE_ZERO_U32: &str = "0u32";
+pub(crate) const MOVE_ZERO_U64: &str = "0u64";
+pub(crate) const MOVE_ZERO_U128: &str = "0u128";
+pub(crate) const MOVE_ZERO_U256: &str = "0u256";
+pub(crate) const MOVE_MAX_U8: &str = "255u8";
```
Comment (vineethk) [inline third_party/move/tools/move-mutator/src/operators/mod.rs:21]: As mentioned in a previous PR, let us please use string conversions of `U8::MAX` etc.

File: third_party/move/tools/move-mutator/src/operators/ifelse.rs [hunk 2]
```diff
@@ -0,0 +1,155 @@
+use crate::operator::{MutantInfo, MutationOperator};
+use crate::report::{Mutation, Range};
+use move_command_line_common::files::FileHash;
+use move_compiler::typing::ast;
+use move_compiler::typing::ast::UnannotatedExp_;
+use std::fmt;
+use std::fmt::Debug;
+
+/// IfElse mutation operator.
+/// Replaces expressions under the if/else statements with literals.
```
Comment (vineethk) [inline third_party/move/tools/move-mutator/src/operators/ifelse.rs:10]:
```suggestion
/// Replaces conditional expressions in if/else statements with literals.
```

File: third_party/move/tools/move-mutator/src/operators/ifelse.rs [hunk 3]
```diff
@@ -0,0 +1,155 @@
+use crate::operator::{MutantInfo, MutationOperator};
+use crate::report::{Mutation, Range};
+use move_command_line_common::files::FileHash;
+use move_compiler::typing::ast;
+use move_compiler::typing::ast::UnannotatedExp_;
+use std::fmt;
+use std::fmt::Debug;
+
+/// IfElse mutation operator.
```
Comment (vineethk) [inline third_party/move/tools/move-mutator/src/operators/ifelse.rs:9]: Another possibility here (either as part of this operator or a new one) is to add have `if (!(expr))` for any `if (expr)`?

**Review by vineethk — COMMENTED** [review]
Again, did not do a thorough review of the mutation operators due to the dependency on `typing`  which should change.

File: third_party/move/tools/move-mutator/src/operators/ifelse.rs (hunk 2 above)
Comment (asmie) [reply third_party/move/tools/move-mutator/src/operators/ifelse.rs:10]: Done.

File: third_party/move/tools/move-mutator/src/operators/ifelse.rs (hunk 3 above)
Comment (asmie) [reply third_party/move/tools/move-mutator/src/operators/ifelse.rs:9]: Yeah, we got it:
```
        // Change if/else expression to true/false.
        let ops: Vec<String> = vec![
            "true".to_owned(),
            "false".to_owned(),
            format!("!({})", cur_op),
        ];
```

File: third_party/move/tools/move-mutator/src/operators/mod.rs (hunk 1 above)
Comment (asmie) [reply third_party/move/tools/move-mutator/src/operators/mod.rs:21]: Done.

**Pull request merged**